### PR TITLE
Improved yaml linting and tightened config. Set YAML 1.2 everywhere

### PR DESCRIPTION
**Pull request by rhfogh**
Companion pull request to mxcubecore PR #957

There may be problems with poetry.lock and/or with mxcubeweb PR #1285.

### CODE REVIEW

File: poetry.lock
Comment (fabcor-maxiv) [inline poetry.lock:1]: This file (`poetry.lock`) should not have changed. I do not understand where this change comes from.

Maybe you could restore the file to its previous state and re-run `poetry install` or something like this.

**Comment by elmjag** [comment]
Same comment as for this PR: https://github.com/mxcube/mxcubecore/pull/957

Let's merge this into `develop`. I don't see a need to intermingle these changes with our XML to YAML conversion work.

**Comment by marcus-oscarsson** [comment]
@rhfogh Did you notice that there is a conflict ?

**Comment by rhfogh** [comment]
@marcus-oscarsson  The conflict should be fixed now. Thanks for pointing it out.

**Pull request merged**